src/R/Editor/Impl/Functions/PackageIndex.cs

```diff
@@ -108,14 +108,17 @@ public sealed class PackageIndex : IPackageIndex {
                     Debug.WriteLine("R function host start: {0} ms", stopwatch.ElapsedMilliseconds);
 
                     // Fetch list of package functions from R session
+                    _disposableBag.ThrowIfDisposed();
```

Comment:
Ideally we should've provide a `CancellationToken` to this method which is linked to both disposing and timeout. But that is a larger change.

Reply:
Yes, I have it with ct in the LS branch.